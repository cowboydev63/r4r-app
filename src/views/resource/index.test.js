import React from 'react';
import { shallow } from 'enzyme';
import { Resource } from './index';

it('renders without crashing', () => {
  shallow(<Resource match={{params: {}}} fetchResource={() => {}}/>);
});
