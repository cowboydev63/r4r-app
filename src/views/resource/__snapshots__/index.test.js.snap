// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource View renders correctly with a resource loaded 1`] = `
<StaticRouter
  basename=""
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ThemeProvider
      theme={Object {}}
    >
      <Resource
        currentResults={Array []}
        fetchResource={[Function]}
        filters={Array []}
        goBack={[Function]}
        history={Array []}
        location={
          Object {
            "search": "",
          }
        }
        match={
          Object {
            "params": Object {
              "id": 1,
            },
          }
        }
        newSearch={[Function]}
        resource={
          Object {
            "body": "<p>R tool for analysis of DNA methylation and expression datasets. Integrative analysis allows reconstruction of in vivo transcription factor networks altered in cancer along with identification of the underlying gene regulatory sequences.</p>",
            "description": "<p>R tool for analysis of DNA methylation and expression datasets.</p>",
            "doCs": Array [
              Object {
                "key": "cbiit",
                "label": "Center for Biomedical Informatics and Informatics Technology (CBIIT)",
              },
            ],
            "id": 1,
            "poCs": Array [
              Object {
                "email": "scotts2@mail.nih.gov",
                "name": Object {
                  "firstName": "Susan",
                  "lastName": "Scott",
                  "middleName": null,
                  "prefix": null,
                  "suffix": null,
                },
                "phone": "240-276-6951",
                "title": "Public Health Advisor, Surveillance Research Program",
              },
              Object {
                "email": "feuerr@mail.nih.gov",
                "name": Object {
                  "firstName": "Eric (Rocky)",
                  "lastName": "Feuer",
                  "middleName": null,
                  "prefix": null,
                  "suffix": null,
                },
                "phone": "240-276-6772",
                "title": "Chief, Statistical Research and Applications Branch, Surveillance Research Program",
              },
            ],
            "researchAreas": Array [
              Object {
                "key": "cancer_omics",
                "label": "Cancer Omics",
              },
            ],
            "researchTypes": Array [
              Object {
                "key": "basic",
                "label": "Basic",
              },
            ],
            "resourceAccess": Object {
              "notes": null,
              "type": "open",
            },
            "title": "Enhancer Linking by Methylation/Expression Relationships (ELMER)",
            "toolSubtypes": Array [
              Object {
                "key": "r_software",
                "label": "R Software",
                "parentKey": "analysis_tools",
              },
            ],
            "toolTypes": Array [
              Object {
                "key": "analysis_tools",
                "label": "Analysis Tools",
              },
            ],
            "website": "http://bioconductor.org/packages/release/bioc/html/ELMER.html",
          }
        }
        searchBarOnChange={[Function]}
      >
        <Theme
          className="r4r-resource"
          element="div"
        >
          <div
            className="r4r-resource  "
          >
            <HelmetWrapper
              defer={true}
              encodeSpecialCharacters={true}
            >
              <SideEffect(NullComponent)
                defer={true}
                encodeSpecialCharacters={true}
                meta={
                  Array [
                    Object {
                      "content": "<p>R tool for analysis of DNA methylation and expression datasets.</p>",
                      "property": "og:description",
                    },
                    Object {
                      "content": "<p>R tool for analysis of DNA methylation and expression datasets.</p>",
                      "name": "description",
                    },
                    Object {
                      "content": "Resources for Researchers: Enhancer Linking by Methylation/Expression Relationships (ELMER) - National Cancer Institute",
                      "property": "twitter:title",
                    },
                    Object {
                      "content": "undefined/resource/1",
                      "property": "og:url",
                    },
                  ]
                }
                title={
                  Array [
                    "Resources for Researchers: ",
                    "Enhancer Linking by Methylation/Expression Relationships (ELMER)",
                    " - National Cancer Institute",
                  ]
                }
                titleAttributes={Object {}}
              >
                <NullComponent
                  defer={true}
                  encodeSpecialCharacters={true}
                  meta={
                    Array [
                      Object {
                        "content": "<p>R tool for analysis of DNA methylation and expression datasets.</p>",
                        "property": "og:description",
                      },
                      Object {
                        "content": "<p>R tool for analysis of DNA methylation and expression datasets.</p>",
                        "name": "description",
                      },
                      Object {
                        "content": "Resources for Researchers: Enhancer Linking by Methylation/Expression Relationships (ELMER) - National Cancer Institute",
                        "property": "twitter:title",
                      },
                      Object {
                        "content": "undefined/resource/1",
                        "property": "og:url",
                      },
                    ]
                  }
                  title={
                    Array [
                      "Resources for Researchers: ",
                      "Enhancer Linking by Methylation/Expression Relationships (ELMER)",
                      " - National Cancer Institute",
                    ]
                  }
                  titleAttributes={Object {}}
                />
              </SideEffect(NullComponent)>
            </HelmetWrapper>
            <Theme
              className="r4r-resource__header"
              element="header"
            >
              <header
                className="r4r-resource__header  "
              >
                <h1>
                  Enhancer Linking by Methylation/Expression Relationships (ELMER)
                </h1>
              </header>
            </Theme>
            <Theme
              className="resource__main"
              element="div"
            >
              <div
                className="resource__main  "
              >
                <Theme
                  className="resource__home"
                  element="div"
                >
                  <div
                    className="resource__home  "
                  >
                    <Link
                      aria-label="Back to home link"
                      replace={false}
                      to="/"
                    >
                      <a
                        aria-label="Back to home link"
                        href="/"
                        onClick={[Function]}
                      >
                        Resources for Researchers Home
                      </a>
                    </Link>
                  </div>
                </Theme>
                <article
                  aria-label="Resource description"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<p>R tool for analysis of DNA methylation and expression datasets. Integrative analysis allows reconstruction of in vivo transcription factor networks altered in cancer along with identification of the underlying gene regulatory sequences.</p>",
                    }
                  }
                />
                <Theme
                  className="resource__link--external"
                  element="div"
                >
                  <div
                    className="resource__link--external  "
                  >
                    <a
                      href="http://bioconductor.org/packages/release/bioc/html/ELMER.html"
                      onClick={[Function]}
                    >
                      Visit Resource
                    </a>
                  </div>
                </Theme>
                <ResourceAccess
                  notes={null}
                  type="open"
                >
                  <article
                    aria-label="Resource Access Information"
                  >
                    <Theme
                      className="resource__access"
                      element="div"
                    >
                      <div
                        className="resource__access  "
                      >
                        <SVG
                          className="r4r-svg-container"
                          iconType="open"
                        >
                          <Theme
                            className="r4r-svg-container"
                            element="div"
                          >
                            <div
                              className="r4r-svg-container  "
                            >
                              <svg
                                height="33px"
                                version="1.1"
                                viewBox="0 0 33 33"
                                width="33px"
                                xmlns="http://www.w3.org/2000/svg"
                                xmlnsXlink="http://www.w3.org/1999/xlink"
                              >
                                <title>
                                  Free Access Icon
                                </title>
                                <path
                                  d="M16.2453077,0.507269231 C7.27311538,0.507269231 0,7.78038462 0,16.7525769 C0,25.7264615 7.27311538,33 16.2453077,33 C25.2191923,33 32.4927308,25.7264615 32.4927308,16.7525769 C32.4927308,7.78038462 25.2191923,0.507269231 16.2453077,0.507269231 Z M3.71334615,12.3965769 L11.957,16.0866538 L14.5381923,17.2188077 L16.4538846,18.0882308 L18.0205385,18.7558462 C18.3095,19.0071538 18.4550385,19.3358846 18.4550385,19.7424615 C18.4550385,20.3225 18.2566154,20.7388077 17.8601923,20.9884231 C17.4633462,21.2414231 16.9844231,21.3670769 16.4238462,21.3670769 C15.0704231,21.3670769 13.8608462,20.8729231 12.7980769,19.888 L10.5638077,22.1501923 C12.0526154,23.4261923 13.6971154,24.0756154 15.4951923,24.0938077 L15.4951923,26.5891154 L17.3808462,26.5891154 L17.3808462,24.0938077 C18.5028462,23.9977692 19.4746538,23.6246154 20.2979615,22.9764615 C21.1174615,22.3287308 21.6365769,21.4927308 21.8481154,20.4663462 L27.8701923,23.1503462 C25.6126538,27.2453077 21.2545385,30.0219615 16.2453077,30.0219615 C8.91761538,30.0219615 2.97803846,24.0806923 2.97803846,16.7525769 C2.97803846,15.2273846 3.23907692,13.7614231 3.71334615,12.3965769 Z M20.8348462,16.5215769 L18.2219231,15.3610769 L16.8008077,14.7235 L14.7416923,13.7944231 C14.6439615,13.6590385 14.5961538,13.5236538 14.5961538,13.3878462 C14.5961538,12.9254231 14.7945769,12.5945769 15.191,12.4029231 C15.5878462,12.2078846 16.0371538,12.1122692 16.5397692,12.1122692 C17.4883077,12.1122692 18.4152692,12.4312692 19.3244615,13.0688462 L21.4432308,10.8925385 C20.2251923,9.96515385 18.8700769,9.46296154 17.3808462,9.38511538 L17.3808462,6.88980769 L15.4951923,6.88980769 L15.4951923,9.38511538 C14.4886923,9.44307692 13.6095385,9.72907692 12.8560385,10.2397308 C12.1008462,10.7537692 11.5885,11.4345 11.3173077,12.2857308 L5.1315,9.51076923 C7.49988462,5.885 11.5914615,3.48530769 16.2453077,3.48530769 C23.5734231,3.48530769 29.513,9.42657692 29.513,16.7525769 C29.513,17.9426923 29.3543462,19.0947308 29.0615769,20.1900769 L20.8348462,16.5215769 Z"
                                  id="Free-icon"
                                />
                              </svg>
                            </div>
                          </Theme>
                        </SVG>
                        <h4>
                          This resource is free
                        </h4>
                      </div>
                    </Theme>
                  </article>
                </ResourceAccess>
                <POCs
                  poCs={
                    Array [
                      Object {
                        "email": "scotts2@mail.nih.gov",
                        "name": Object {
                          "firstName": "Susan",
                          "lastName": "Scott",
                          "middleName": null,
                          "prefix": null,
                          "suffix": null,
                        },
                        "phone": "240-276-6951",
                        "title": "Public Health Advisor, Surveillance Research Program",
                      },
                      Object {
                        "email": "feuerr@mail.nih.gov",
                        "name": Object {
                          "firstName": "Eric (Rocky)",
                          "lastName": "Feuer",
                          "middleName": null,
                          "prefix": null,
                          "suffix": null,
                        },
                        "phone": "240-276-6772",
                        "title": "Chief, Statistical Research and Applications Branch, Surveillance Research Program",
                      },
                    ]
                  }
                >
                  <article>
                    <h2>
                      Contact Information
                    </h2>
                    <ContactInformation
                      contact={
                        Object {
                          "email": "scotts2@mail.nih.gov",
                          "name": Object {
                            "firstName": "Susan",
                            "lastName": "Scott",
                            "middleName": null,
                            "prefix": null,
                            "suffix": null,
                          },
                          "phone": "240-276-6951",
                          "title": "Public Health Advisor, Surveillance Research Program",
                        }
                      }
                      key="0"
                    >
                      <Theme
                        className="contact-information"
                        element="address"
                      >
                        <address
                          className="contact-information  "
                        >
                          <p>
                             Susan  Scott 
                          </p>
                          <p>
                            Public Health Advisor, Surveillance Research Program
                          </p>
                          <p>
                            240-276-6951
                          </p>
                          <a
                            href="mailto:scotts2@mail.nih.gov"
                          >
                            scotts2@mail.nih.gov
                          </a>
                        </address>
                      </Theme>
                    </ContactInformation>
                    <ContactInformation
                      contact={
                        Object {
                          "email": "feuerr@mail.nih.gov",
                          "name": Object {
                            "firstName": "Eric (Rocky)",
                            "lastName": "Feuer",
                            "middleName": null,
                            "prefix": null,
                            "suffix": null,
                          },
                          "phone": "240-276-6772",
                          "title": "Chief, Statistical Research and Applications Branch, Surveillance Research Program",
                        }
                      }
                      key="1"
                    >
                      <Theme
                        className="contact-information"
                        element="address"
                      >
                        <address
                          className="contact-information  "
                        >
                          <p>
                             Eric (Rocky)  Feuer 
                          </p>
                          <p>
                            Chief, Statistical Research and Applications Branch, Surveillance Research Program
                          </p>
                          <p>
                            240-276-6772
                          </p>
                          <a
                            href="mailto:feuerr@mail.nih.gov"
                          >
                            feuerr@mail.nih.gov
                          </a>
                        </address>
                      </Theme>
                    </ContactInformation>
                  </article>
                </POCs>
                <DOCs
                  doCs={
                    Array [
                      Object {
                        "key": "cbiit",
                        "label": "Center for Biomedical Informatics and Informatics Technology (CBIIT)",
                      },
                    ]
                  }
                >
                  <Theme
                    aria-label="NCI Affiliation Information"
                    className="resource__docs"
                    element="article"
                  >
                    <article
                      aria-label="NCI Affiliation Information"
                      className="resource__docs  "
                    >
                      <h2>
                        NCI Affiliation
                      </h2>
                      This resource is managed by the Center for Biomedical Informatics and Informatics Technology (CBIIT).
                    </article>
                  </Theme>
                </DOCs>
              </div>
            </Theme>
            <Theme
              className="resource__nav"
              element="div"
            >
              <div
                className="resource__nav  "
              >
                <section
                  role="search"
                >
                  <SearchBar
                    onChange={[Function]}
                    onSubmit={[Function]}
                    page="resource"
                    placeholder="Search resources"
                  >
                    <Theme
                      className="searchbar__container"
                      element="form"
                      onSubmit={[Function]}
                    >
                      <form
                        className="searchbar__container  "
                        onSubmit={[Function]}
                      >
                        <input
                          aria-label="Input search text here to search for a resource"
                          onChange={[Function]}
                          placeholder="Search resources"
                          type="text"
                        />
                        <Theme
                          aria-label="submit search"
                          className="searchbar__button--submit"
                          element="button"
                          type="submit"
                        >
                          <button
                            aria-label="submit search"
                            className="searchbar__button--submit  "
                            type="submit"
                          >
                            <SVG
                              className="r4r-svg-container"
                              iconType="magnify"
                            >
                              <Theme
                                className="r4r-svg-container"
                                element="div"
                              >
                                <div
                                  className="r4r-svg-container  "
                                >
                                  <svg
                                    height="16px"
                                    version="1.1"
                                    viewBox="0 0 16 16"
                                    width="16px"
                                    xmlns="http://www.w3.org/2000/svg"
                                    xmlnsXlink="http://www.w3.org/1999/xlink"
                                  >
                                    <title>
                                      Magnifying Glass Icon
                                    </title>
                                    <defs />
                                    <g
                                      fill="none"
                                      fillRule="evenodd"
                                      id="r4r-magnify-icon"
                                      stroke="none"
                                      strokeWidth="1"
                                    >
                                      <path
                                        d="M12.370756,11.6338533 L11.497863,11.1003191 L12.0511346,10.2398009 C12.6676075,9.28098399 13,8.16636264 13,7 C13,3.6862915 10.3137085,1 7,1 C3.6862915,1 1,3.6862915 1,7 C1,10.3137085 3.6862915,13 7,13 C8.18648905,13 9.31933446,12.6559913 10.2886875,12.0193666 L11.1435654,11.4579236 L11.685632,12.3252172 C11.7040473,12.3546813 11.7258067,12.3825474 11.7510134,12.4086496 L14.5296468,15.2860088 C14.7214715,15.484649 15.0380057,15.4901741 15.2366459,15.2983495 C15.4352861,15.1065249 15.4408113,14.7899906 15.2489866,14.5913504 L12.4703532,11.7139912 C12.4398346,11.6823883 12.4065292,11.6557187 12.370756,11.6338533 Z"
                                        id="Combined-Shape"
                                        stroke="#FFFFFF"
                                        strokeWidth="2"
                                      />
                                    </g>
                                  </svg>
                                </div>
                              </Theme>
                            </SVG>
                          </button>
                        </Theme>
                      </form>
                    </Theme>
                  </SearchBar>
                </section>
                <nav>
                  <h2>
                    Find Related Resources
                  </h2>
                  <Theme
                    className="similar-resource__container"
                    element="div"
                  >
                    <div
                      className="similar-resource__container  "
                    />
                  </Theme>
                </nav>
              </div>
            </Theme>
          </div>
        </Theme>
      </Resource>
    </ThemeProvider>
  </Router>
</StaticRouter>
`;

exports[`Resource View renders correctly with no resource 1`] = `<CTS_Spinner />`;
